### PR TITLE
[BugFix] Fix complie error in staros

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Fix compile error if enable --use-staros

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [x] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/storage/lake/update_manager.cpp [hunk 1]
```diff
@@ -41,6 +41,10 @@ UpdateManager::UpdateManager(LocationProvider* location_provider, MemTracker* me
 
     _index_cache.set_mem_tracker(_index_cache_mem_tracker.get());
     _update_state_cache.set_mem_tracker(_update_state_mem_tracker.get());
+
+    int64_t byte_limits = ParseUtil::parse_mem_spec(config::mem_limit, MemInfo::physical_mem());
+    int32_t update_mem_percent = std::max(std::min(100, config::update_memory_limit_percent), 0);
```
Comment (luohaha) [inline be/src/storage/lake/update_manager.cpp:46]: call `update_primary_index_memory_limit`?

File: be/src/storage/lake/update_manager.h [hunk 2]
```diff
@@ -98,9 +100,18 @@ class UpdateManager {
...
     void expire_cache();
 
+    void evict_cache(int64_t memory_urgent_level, int64_t memory_high_level);
+
     // check if pk index's cache ref == ref_cnt
     bool TEST_check_primary_index_cache_ref(uint32_t tablet_id, uint32_t ref_cnt);
 
+    Status update_primary_index_memory_limit(int32_t update_memory_limit_percent) {
+        int64_t byte_limits = ParseUtil::parse_mem_spec(config::mem_limit, MemInfo::physical_mem());
+        int32_t update_mem_percent = std::max(std::min(100, update_memory_limit_percent), 0);
```
Comment (luohaha) [inline be/src/storage/lake/update_manager.h:110]: Just `int32_t update_mem_percent = std::min(100, update_memory_limit_percent)`?

**Review by kevincai — APPROVED** [review]
good on compiling.

**Review by sduzh — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**